### ISSUE TITLE
Move to `bitcoind` crate for integration testing

### ISSUE THREAD

**Issue by sanket1729** [rust-bitcoin/rust-miniscript#361]
`bitcoind` crate is a more maintained wrapper around bitcoin-core-rpc. We should shift our integration testing framework to use it. It fits it nicely with cargo tests and allows locally running integration tests easily. 

I think we should mark that as optional in the CI build, as it will delay our release cycle adding yet another dependency that needs to be updated every-time there is a rust-bitcoin release.

**Comment by apoelstra** [comment]
To clarify: when rust-bitcoin updates, both this crate and `bitcoind` will need to update their rust-bitcoin dependency, so your proposal is to make the dep optional (and disable/ignore it) during the periods when we're waiting for `bitcoind` to catch up?

**Comment by sanket1729** [comment]
Make this check optional in CI build. 

For every release in the past few releases, we have been stuck for 1-2 weeks after rust-bitcoin to have a rust-miniscript release out because of this. Ideally, we want this in the CI check. I am about to suggest adding more maintainers to `rust-bitcoincore-rpc`. Basically only steven maintains this now, and he is busy with other multiple projects. 

It is a fairly low amount of work to keep updating this every rust-bitcoin release(took me 10 mins for this PR) https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/217. 

If we also integrate this with the rust-bitcoin cycle and add rust-bitcoin maintainers there, it can help speed up things for us here. I will ask @stevenroose to see if he is okay with that. We don't need the code to be high quality in `rust-bitcoincore-rpc`, so maintenance is just chasing obvious compiler errors every rust-bitcoin release and making a prompt release.

**Comment by sanket1729** [comment]
I realized that there are two related things in this issue. Created a separate issue for it.

1) Move to bitcoind. This issue
2) Fix CI failure on merge commit, blocked on rust-bitcoincore-rpc release.
